src/app/modules/analytics-config/components/edit-histogram-label/edit-histogram-label.component.ts

```diff
@@ -0,0 +1,229 @@
+            this.enableXUnitField();
+          }
+        });
+      combineLatest([
+        this.yAxisConfig.dataFieldValue.valueChanges,
+        this.yAxisConfig.dataFieldFunction.valueChanges
+      ])
+        .pipe(
+          takeUntil(this._destroyed$),
+          distinctUntilChanged()
+        )
+        .subscribe(_ => {
+          if (this.metricCollectionIsCount()) {
+            this.disableYUnitField();
+            this.setYUnitFieldWithCollectionUnit();
+          } else {
+            this.enableYUnitField();
+          }
+        });
+
+      this.dataStep.collection.valueChanges
+        .pipe(
+          takeUntil(this._destroyed$),
+          distinctUntilChanged()
+        ).subscribe(value => {
+          if (this.metricCollectionIsCount()) {
+            this.disableYUnitField();
+            this.setYUnitFieldWithCollectionUnit();
+          }
+        });
+    }
+
+    private getLookAndFeelControl() {
+      if (!this.lookAndFeelFormControl || this.currentCollectionName !== this.dataStep.collection.value) {
+        this.currentCollectionName = this.dataStep.collection.value;
+        const globalFormGroup = this.mainFromService.lookAndFeelConfig.getGlobalFg();
+        const lookAndFeelFormControl = (<FormArray>globalFormGroup.customControls.units.value).controls
+          .filter(c => c.value.collection === this.dataStep.collection.value);
+        this.lookAndFeelFormControl = lookAndFeelFormControl[0] ?? null;
+      }
+    }
+
+    private xFieldIsDate() {
+      return this.dataStep.aggregation.get('aggregationFieldType').value === 'time';
+    }
+
+    private metricCollectionIsCount() {
+      return this.yAxisConfig.dataFieldFunction.value === 'Count';
+    }
+
+    private disableXUnitField() {
+      this.xAxisConfig.readonly.next(true);
+      this.xAxisConfig.unitControl.setValue('');
+      this.xAxisConfig.message.next('Managed by Arlas');
+    }
+
+    private disableYUnitField() {
+      this.yAxisConfig.readonly.next(true);
+      this.yAxisConfig.message.next('Field set by unit collection in look and feel');
+    }
+
+    private setYUnitFieldWithCollectionUnit() {
+      this.getLookAndFeelControl();
+      const collectionUnit = (this.lookAndFeelFormControl) ? this.lookAndFeelFormControl?.value.unit : '';
+      this.yAxisConfig.unitControl.setValue(collectionUnit);
+    }
+
+    private enableXUnitField(value?: string) {
+      this.xAxisConfig.readonly.next(false);
+      this.xAxisConfig.labelControl.setValue('');
+      this.xAxisConfig.message.next('');
+      if (value) {
+        this.xAxisConfig.unitControl.setValue(marker(value));
+      } else {
+        this.xAxisConfig.unitControl.setValue('');
+      }
+    }
+
+    private enableYUnitField(value?: string) {
+      this.yAxisConfig.readonly.next(false);
+      this.yAxisConfig.labelControl.setValue('');
+      this.yAxisConfig.message.next('');
+      if (value) {
+        this.yAxisConfig.unitControl.setValue(marker(value));
```

Comment:
The marker here is not useful as it just grabs hardcoded strings and members of enums when the parameter of the function is an enum